Source/JoystickPlugin/Public/Widgets/SJoystickInputSelector.cpp

```diff
@@ -83,6 +84,12 @@ FReply SJoystickInputSelector::OnKeyUp(const FGeometry& MyGeometry, const FKeyEv
 
 FReply SJoystickInputSelector::OnAnalogValueChanged(const FGeometry& MyGeometry, const FAnalogInputEvent& InAnalogInputEvent)
 {
+	/** Don't process events in dead zone */
+	if (InAnalogInputEvent.GetAnalogValue() <= -1.0f+(1.0f*DeadZonePercentage/100) || InAnalogInputEvent.GetAnalogValue() >= 1.0f-(1.0f*DeadZonePercentage/100) || InAnalogInputEvent.GetAnalogValue() >= 0.0f-(1.0f*DeadZonePercentage/100) || InAnalogInputEvent.GetAnalogValue() >= 0.0f+(1.0f*DeadZonePercentage/100))
```

Comment:
My feedback would be to do an `abs` on the GetAnalog value and then check the deadzone
I would also suggest changing `DeadZonePercentage` to a float between 0 and 1
This would simplify the if statement to something like this:
```cpp
const float AbsAnalogValue = FMath::Abs(InAnalogInputEvent.GetAnalogValue());
if (AbsAnalogValue >= DeadZone)
{
	return SCompoundWidget::OnAnalogValueChanged(MyGeometry, InAnalogInputEvent);
}
```

Reply:
I just pushed an improved version thanks to your review.